source/renderer/app/api/ada/index.js

```diff
@@ -250,21 +250,37 @@ export default class AdaApi {
     const { walletId, skip, limit } = request;
 
     const accounts: AdaAccounts = await getAdaWalletAccounts(this.config, { walletId });
-    const accountIndex = accounts[0].index;
-    const page = skip === 0 ? 1 : (skip / limit) + 1;
-    const perPage = limit > 50 ? 50 : limit;
 
     const params = {
-      accountIndex,
-      page,
-      per_page: perPage,
+      accountIndex: accounts[0].index,
+      page: skip === 0 ? 1 : (skip / limit) + 1,
+      per_page: limit > 50 ? 50 : limit,
       wallet_id: walletId,
       sort_by: 'DES[created_at]',
     };
 
+    // Calculate how many pages we need to load
+    const pageLimit = Math.ceil(limit / params.per_page);
+
     try {
-      const history: AdaTransactionsV1 = await getAdaHistoryByWallet(this.config, params);
-      const transactions = history.map(data => _createTransactionFromServerDataV1(data));
+      const {
+        data: history,
+        meta
+      }: AdaTransactionsV1 = await getAdaHistoryByWallet(this.config, params);
+      const { totalPages } = meta.pagination;
+
+      // In case there is more than one page,
+      // it loops through them and adds to the `history` array
+      if (totalPages > 1 && limit > 50) {
```

Comment:
You could remove the comment by using a descriptive var like `const hasMultiplePages = (totalPages > 1 && limit > perPage)` in the conditional instead of explaining what the expression does 😉 

(in general if you find yourself explaining simple things like this in comments, try to find a more descriptive way to express the same idea in code)